File: src/javascript/javascript-dsa/algorithm/sorting/basic/insertion-sort.js
/*
    In-Place & Stable
    Runtime: O(N^2)
*/
function insertionSort(unsortedList) {
    for (var i = 0; i < unsortedList.length; i++) {
        let current = i;
        // gets the smallest element and inserts at current index
        while (current > 0 && unsortedList[current] < unsortedList[current - 1]) {
            const temp = unsortedList[current];
            // swaps current smaller element with the element before it
            unsortedList[current] = unsortedList[current - 1];
            unsortedList[current - 1] = temp;
            current--;
        }
    }
    return unsortedList;
}




const arr = [65, 99, 10, 100,1, 200, 47, 23]



console.log("Unsorted", arr)
const result = insertionSort(arr);
console.log("Sorted", arr);

File: src/python/algorithms/sorting/insertion-sort.py
def insertion_sort(arr):
    for i in range(len(arr)):
        current = i

        while current > 0 and arr[current] < arr[current - 1]:
            arr[current], arr[current - 1] = arr[current - 1], arr[current]
            current -= 1
    return arr


arr = [65, 99, 10, 100, 1, 200, 47, 23]
# print(arr)

insertion_sort(arr)

# print(arr)


def insertion_sort_practice(arr):

    return arr
